### PR TITLE
enhancement(elasticsearch sink): Add option to specify a default index if template cannot be resolved for Elasticsearch destination index

### PR DESCRIPTION
**Pull request by ArunPiduguDD**
### Summary

Currently, the Elasticsearch destionation has the option to specify an [index](https://vector.dev/docs/reference/configuration/sinks/elasticsearch/#bulk.index) to write events to. This field supports template syntax, however if the template is not able to be resolved, the default behavior is to drop the event.

Adding an optional field that specifies a fallback index that logs can be written to if there is a template rendering error.

### CODE REVIEW

**Review by neko-dd — APPROVED** [review]

**Comment by pront** [comment]
let's wait for a stamp from one more reviewer (not docs) before we merge this one

File: src/sinks/elasticsearch/mod.rs
Comment (jszwedko) [inline src/sinks/elasticsearch/mod.rs:201]: We typically only emit error level events in sinks when they are non-recoverable. Here it is falling back to the configured default index. I think we can just drop this emit since if users are configuring a fallback index they are expecting some events to fail and so there isn't really anything to alert them to.

I could see some users wanting a metric about how many events failed to render the index, but likely we don't need a log. I'll leave it up to you if you want to add a metric. We could wait for someone to ask for it.

File: changelog.d/add_default_fallback_index_elasticsearch_destination.enhancement.md
Comment (jszwedko) [inline changelog.d/add_default_fallback_index_elasticsearch_destination.enhancement.md:1]:
```suggestion
Add an option to Elasticsearch sink to set a fallback index if the provided template in the `bulk.index` field
```

**Review by jszwedko — COMMENTED** [review]
Just one comment, otherwise this looks good to me.

File: src/sinks/elasticsearch/mod.rs
Comment (ArunPiduguDD) [reply src/sinks/elasticsearch/mod.rs:201]: Intended use-case for this is to ensure logs are not dropped, but also inform the user if a field they are expecting to be present in a log is not there. Wanted to keep this `emit` to make sure the user is kept informed in these scenarios

Comment (jszwedko) [reply src/sinks/elasticsearch/mod.rs:201]: Gotcha, I see. I didn't realize that `TemplateRenderingError` is already emitting as a warning if events aren't dropped so 👍  to rolling with this.

**Review by jszwedko — APPROVED** [review]

**Pull request merged**
